contracts/bonding/BondingCheckpoints.sol

```diff
@@ -0,0 +1,351 @@
+         */
+        uint256 bondedAmount;
+        /**
+         * @dev The address of the delegate the account is bonded to. In case of transcoders this is their own address.
+         */
+        address delegateAddress;
+        /**
+         * @dev The amount of tokens delegated from delegators to this account. This is only set for transcoders, which
+         * have to self-delegate first and then have tokens bonded from other delegators.
+         */
+        uint256 delegatedAmount;
+        /**
+         * @dev The last round during which the delegator claimed its earnings. This pegs the value of bondedAmount for
+         * rewards calculation in {BondingManager-delegatorCumulativeStakeAndFees}.
+         */
+        uint256 lastClaimRound;
+        /**
+         * @dev The last round during which the transcoder called {BondingManager-reward}. This is needed to find a
+         * reward pool for any round when calculating historical rewards.
+         *
+         * Notice that this is the only field that comes from the Transcoder struct in BondingManager, not Delegator.
+         */
+        uint256 lastRewardRound;
+    }
+
+    /**
+     * @dev Stores a list of checkpoints for an account, queryable and mapped by start round. To access the checkpoint
+         for a given round, find the checkpoint with the highest start round that is lower or equal to the queried round
+        ({SortedArrays-findLowerBound}) and then fetch the specific checkpoint on the data mapping.
+        */
+    struct BondingCheckpointsByRound {
+        uint256[] startRounds;
+        mapping(uint256 => BondingCheckpoint) data;
+    }
+
+    /**
+     * @dev Checkpoints by account (delegators and transcoders).
+     */
+    mapping(address => BondingCheckpointsByRound) private bondingCheckpoints;
+
+    /**
+     * @dev Rounds in which we have checkpoints for the total active stake. This and {totalActiveStakeCheckpoints} are
+     * handled in the same wat that {BondingCheckpointsByRound}, with rounds stored and queried on this array and
+     * checkpointed value stored and retrieved from the mapping.
+     */
+    uint256[] totalStakeCheckpointRounds;
+    /**
+     * @dev See {totalStakeCheckpointRounds} above.
+     */
+    mapping(uint256 => uint256) private totalActiveStakeCheckpoints;
+
+    // IERC6372 interface implementation
+
+    /**
+     * @notice Clock is set to match the current round, which is the checkpointing
+     *  method implemented here.
+     */
+    function clock() public view returns (uint48) {
+        return SafeCast.toUint48(roundsManager().currentRound());
+    }
+
+    /**
+     * @notice Machine-readable description of the clock as specified in EIP-6372.
+     */
+    // solhint-disable-next-line func-name-mixedcase
+    function CLOCK_MODE() public pure returns (string memory) {
+        // TODO: Figure out the right value for this
+        return "mode=livepeer_round&from=default";
```

Comment:
After reading through [this](https://eips.ethereum.org/EIPS/eip-6372#clock_mode):
> If operating using any other mode, then this function SHOULD return a unique identifier for the encoded mode field.

I'll just change this to `mode=livepeer_round` (same thing without "from")